cli/src/build.rs

```diff
@@ -148,6 +149,16 @@ pub fn build_project(arg_mode: &Option<String>, arg_platforms: &Option<Vec<Strin
 
     // Architecture selection for iOS or Android
     let selected_architectures = platform.select_archs(&mut config);
+    let mut ios_hash = HashSet::new();
+    let mut android_hash = HashSet::new();
+    for arch in selected_architectures.get("ios").unwrap() {
+        ios_hash.insert(arch.clone());
+    }
+    for arch in selected_architectures.get("android").unwrap() {
+        android_hash.insert(arch.clone());
+    }
+    config.ios = Some(ios_hash);
```

Comment:
Wondering why updating `ios` and `android` here instead of updating inside `select_archs` as original implementation?

I think the code is clearer if we move these update logic inside `select_archs()` and I did the same for `select()` to update `target_platforms`. If you prefer to move them outside, then I would suggest
1. do the same thing to `select()` (move `target_platforms` out)
2. try to remvoe `mut` for `config` when calling `select_archs()`, `mut` should not be necessary

Reply:
right it doesn't have to be written again
I removed them all [029aaec](https://github.com/zkmopro/mopro/pull/417/commits/029aaec3858a5897fbbc747d4efea1752807973e)